### PR TITLE
Billing: Add direction to billing details

### LINKED ISSUE #31681: Billing: Not clear that receipts can be edited
60vphk-1

There is no indication that you can add your own details to the receipt.

People often contact support to ask for a receipt with VAT information or some other special piece of info. The solution is to add the info to the "Billing Details" section in the receipt before printing. But if you don’t already know you can do it, it’s really hard to discover that functionality on your own.

### PR DESCRIPTION
**Pull request by sixhours**
#### Changes proposed in this Pull Request

* Adds descriptive text to the billing details section, giving the user indication they can add their own information to the receipt before printing.
* Updates the editable DIV to a Textarea for better accessibility
* The descriptive text will not be shown on the printed receipt.

**Before**

Screen Shot 2019-03-28 at 11 32 28 AM

**After**

Screen Shot 2019-04-02 at 1 19 06 PM

#### Testing instructions

* Switch to this PR, navigate to `/me/purchases/billing/` and click on an individual purchase
* You should see the descriptive text under the Billing Details header, and the `div` should be a `textarea` instead .

Fixes #31681

### CODE REVIEW

**Comment by scruffian** [comment]
I made a few changes to this approach to get round a warning in React and keep the code simple.

I have tested in IE11 and it works well there.

File: client/me/billing-history/style.scss [hunk 1]
```diff
@@ -192,6 +192,18 @@
 	min-height: 1.2em;
 }
 
+.billing-history__billing-details-editable {
+	&:hover {
+		outline: 1px black dashed;
```
Comment (scruffian) [inline client/me/billing-history/style.scss]: I changed this to an outline so that the text doesn't move around when the box is focussed.

**Comment by sixhours** [comment]
My concern with using the `content` attribute as a placeholder is whether it's accessible to folks with screen readers. My research shows that it probably isn't for folks using IE 11 with a screen reader, and WCAG guidelines suggest it's not accessible: https://www.w3.org/TR/WCAG20-TECHS/F87.html

I'm not sure how to make it fully accessible and placeholder-like without switching to a `textarea`, though.

**Comment by scruffian** [comment]
Can we deal with that by also adding an aria-label?

**Comment by enriquesanchez** [comment]
There are contradicting opinions about placeholder text and its accessibility. Here's a really good article against it (with suggestions at the end on what to do instead): https://www.smashingmagazine.com/2018/06/placeholder-attribute/

And here's a more relaxed approach that suggests that placeholder text can be used as long as we provide additonal labels for screen readers: https://a11yproject.com/posts/placeholder-input-elements/

I think there's a couple small things we can do to improve that field:

1. Make 'Click to add your billing details" look more like an input field, or if we want to hide the field by default then make it look like a link or button and then show the field. 
2. Add an aria-label or aria-labelledby (in addition to placeholder text) to provide assistive text.

I honestly really like the solution suggested on that first article, I think this is very clean and obvious:
solution

**Comment by sixhours** [comment]
@enriquesanchez Thanks for taking a look! I like that first solution, too. Something like this, which better describes the field's function, might be both more accessible and understandable:

Screen Shot 2019-04-02 at 9 44 39 AM

Thoughts @scruffian ?

**Comment by scruffian** [comment]
Looks good. I wonder if we should also convert this into a textarea for accessibility reasons?

**Comment by sixhours** [comment]
It does seem like we've come full circle to using a textarea after all, huh? :D

**Comment by sixhours** [comment]
I experimented with `TextareaAutosize` but it sets a 92px minimum height as an inline style regardless of the number of rows specified, so it looks pretty funky:

Screen Shot 2019-04-02 at 11 54 11 AM

I think we can get away with using a DIV since it looks like an input, and I've also added an `aria-describedby` attribute pointing to the descriptive text above it.

**Comment by enriquesanchez** [comment]
>I think we can get away with using a DIV since it looks like an input

@sixhours If there's a way to make this an actual text field/area instead of a DIV that would so much better for a11y in general.

**Comment by sixhours** [comment]
Figured out the sizing problem! I've updated this to use an autosizing Textarea; could you take a look at the code and let me know if I'm using my labels/aria-labelledby correctly @enriquesanchez ? Many thanks!

Screen Shot 2019-04-02 at 1 19 06 PM

**Comment by enriquesanchez** [comment]
>could you take a look at the code and let me know if I'm using my labels/aria-labelledby correctly?

@sixhours Looks good to me! 👍

File: client/me/billing-history/receipt.jsx [hunk 2]
```diff
@@ -138,8 +139,23 @@ class BillingReceipt extends React.Component {
 
 		return (
 			<li className="billing-history__billing-details">
-				<strong>{ translate( 'Billing Details' ) }</strong>
-				<div contentEditable="true" />
+				<label for="billing-history__billing-details-textarea">
```
Comment (scruffian) [inline client/me/billing-history/receipt.jsx]: `for` should be `htmlFor`

**Pull request merged**